src/lightning_app/runners/cloud.py

```diff
@@ -293,6 +280,15 @@ def dispatch(
             queue_server_type = V1QueueServerType.REDIS if CLOUD_QUEUE_TYPE == "redis" else V1QueueServerType.HTTP
             if find_instances_resp.lightningapps:
                 existing_instance = find_instances_resp.lightningapps[0]
+
+                # TODO: support multiple instances / 1 instance per cluster
+                if existing_instance.spec.cluster_id != cluster_id:
+                    raise ValueError(
+                        f"Can not start app '{name}' on cluster '{cluster_id}' "
+                        f"since app already exists on '{existing_instance.spec.cluster_id}'. "
+                        "To run this app on another cluster, give it another name with the --name flag"
```

Comment:
```suggestion
                        f"Can not start app '{name}' on cluster '{cluster_id}' "
                        f"since this app already exists on '{existing_instance.spec.cluster_id}'. "
                        "To run it on another cluster, give it a new name with the --name option."
```
Nit: Avoiding word repetition and --name is not strictly speaking a flag :))

Reply:
I like the suggestion!